casbin/util/builtin_operators.py

```diff
@@ -194,6 +194,33 @@ def key_match4_func(*args) -> bool:
     return key_match4(name1, name2)
 
 
+def key_match5(key1: str, key2: str) -> bool:
+    """
+    key_match5 determines whether key1 matches the pattern of key2 (similar to RESTful path), key2 can contain a *
+    For example,
+    - "/foo/bar?status=1&type=2" matches "/foo/bar"
+    - "/parent/child1" and "/parent/child1" matches "/parent/*"
+    - "/parent/child1?status=1" matches "/parent/*"
+    """
+    i = key1.find("?")
+    if i != -1:
+        key1 = key1[:i]
+
+    key2 = key2.replace("/*", "/.*")
+
+    re_pattern = r"{[^/]+}"
+    key2 = re.sub(re_pattern, r"[^/]+", key2)
+
+    return regex_match(key1, "^" + key2 + "$")
```

Comment:
@BustDot Please pre-compile the regular expression at the beginning of the file to improve efficiency.

Reply:
Resolved.